src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc

```diff
@@ -60,7 +60,7 @@ void SourceGeneratorBase::WriteGeneratedCodeAttributes(io::Printer* printer) {
 }
 
 std::string SourceGeneratorBase::class_access_level() {
-  return (IsDescriptorProto(descriptor_) || this->options()->internal_access) ? "internal" : "public";
+  return this->options()->internal_access ? "internal" : "public";
```

Comment:
I don't think we can make the generated descriptor file to public now because proto2 features are not well tested. Can this change hold until the tests are added and proto2 gen is public?

Reply:
Sure.

Reply:
I reintroduced the internal access level for the descriptor proto. There were changes I manually made to generated test code to make sure the access levels matched up. I also commented out the CustomOptions deprecation for all the descriptors. The commented descriptor GetOption methods will be uncommented with the next PR.